include/solvers/mpm_base.tcc

```diff
@@ -96,10 +96,44 @@ mpm::MPMBase<Tdim>::MPMBase(const std::shared_ptr<IO>& io) : mpm::MPM(io) {
   try {
     if (post_process_.at("vtk_statevars").is_array() &&
         post_process_.at("vtk_statevars").size() > 0) {
-      for (unsigned i = 0; i < post_process_.at("vtk_statevars").size(); ++i) {
-        std::string attribute =
-            post_process_["vtk_statevars"][i].template get<std::string>();
-        vtk_statevars_.emplace_back(attribute);
+      // Iterate over state_vars
+      for (const auto& svars : post_process_["vtk_statevars"]) {
+        // Initial phase id and state vars
+        unsigned phase_id = 0;
+        std::vector<std::string> state_var;
+
+        // If state variables are arranged as string
+        if (svars.is_string()) {
+          std::string attribute = svars.template get<std::string>();
+          state_var.emplace_back(attribute);
+        }
+        // If state variables are arranged as json object
+        else if (svars.is_object()) {
+          // Phase id
+          if (svars.contains("phase_id"))
+            phase_id = svars.at("phase_id").template get<unsigned>();
+
+          // State variables
+          if (svars.at("statevars").is_array() &&
+              svars.at("statevars").size() > 0) {
+            for (unsigned i = 0; i < svars.at("statevars").size(); ++i) {
+              std::string attribute =
+                  svars["statevars"][i].template get<std::string>();
+              state_var.emplace_back(attribute);
+            }
+          } else {
+            throw std::runtime_error(
+                "No VTK statevariable were specified, none will be generated");
+          }
+        }
+
+        // Check if vtk_statevars_ has the designated phase_id
```

Comment:
I don't follow what's happening here. Why can't we check before we add to `state_var`?